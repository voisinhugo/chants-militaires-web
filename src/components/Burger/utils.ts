import songs from "../../data";

export const getRandomSong = () =>
  Math.floor(Math.random() * Object.keys(songs).length);
