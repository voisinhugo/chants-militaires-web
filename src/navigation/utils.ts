const BASE_PATH = "/chants-militaires-web";

export const PATHS = {
  song: "/song"
};

export const getPath = (path: string = "") => `${BASE_PATH}${path}`;
