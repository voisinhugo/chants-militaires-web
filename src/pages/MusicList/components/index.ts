export { MusicItem } from "./MusicItem";
export { SearchBar } from "./SearchBar";
