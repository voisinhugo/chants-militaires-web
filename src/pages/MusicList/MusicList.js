import React from "react";
import styled from "styled-components";
import { MusicItem } from "./components";
import songs from "../../data";
import theme from "../../theme";

const sortedSongs = Object.values(songs).sort((a, b) => a.title > b.title);

const Container = styled.div`
  background-color: ${theme.color.background};
`;

export const MusicList = ({ navigation }) => {
  return (
    <Container>
      <ul>
        {sortedSongs.map(item => (
          <MusicItem key={item.id.toString()} item={item} />
        ))}
      </ul>
    </Container>
  );
};
