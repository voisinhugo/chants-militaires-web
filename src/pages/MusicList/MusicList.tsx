import React, { FunctionComponent } from "react";
import styled from "styled-components";
import { MusicItem } from "./components";
import songs from "../../data";
import theme from "../../theme";
import { Header } from "../../components";
import { Song } from "../../react-app-env";
import { HEADER_HEIGHT } from "../../components/Header";

const sortedSongs = Object.values(songs).sort((a: Song, b: Song) =>
  a.title > b.title ? 1 : -1
);

const ListContainer = styled.div`
  background-color: ${theme.color.background};
  padding-top: ${theme.margin.x1 + HEADER_HEIGHT}px;
  padding-bottom: ${theme.margin.x1}px;
`;

export const MusicList: FunctionComponent = () => {
  return (
    <>
      <Header title="Liste des chants" />
      <ListContainer>
        {sortedSongs.map((item: Song) => (
          <MusicItem key={item.id.toString()} song={item} />
        ))}
      </ListContainer>
    </>
  );
};
