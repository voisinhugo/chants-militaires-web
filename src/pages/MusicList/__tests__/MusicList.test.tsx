import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { MusicList } from "../MusicList";
import { renderPage } from "../../../tests/utils";

const song0Id = 0;
const song0Title = "toto";
jest.mock("../../../data/index", () => ({
  "0": { id: 0, title: "toto" },
  "1": { id: 1, title: "tata" }
}));

const mockHistoryPush = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    push: mockHistoryPush
  })
}));

describe("MusicList", () => {
  it("should render", () => {
    const musicList = renderPage(<MusicList />);

    expect(musicList).toMatchSnapshot();
  });
  it("should navigate to lyric page", () => {
    const { getByText } = render(<MusicList />);

    const itemButton = getByText(song0Title);
    fireEvent.click(itemButton);

    const expectedPath = `/chants-militaires-web/song/${song0Id}`;
    expect(mockHistoryPush).toHaveBeenCalledWith(expectedPath);
  });
});
