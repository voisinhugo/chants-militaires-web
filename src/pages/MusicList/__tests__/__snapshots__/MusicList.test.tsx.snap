// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MusicList should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AykKG eNPyVs"
      >
        <div>
          <div
            class="bm-overlay"
            style="position: fixed; z-index: 1000; width: 100%; height: 100%; background: rgba(0, 0, 0, 0.3); opacity: 0; transform: translate3d(100%, 0, 0); transition: opacity 0.3s, transform 0s 0.3s;"
          />
          <div
            class="bm-menu-wrap"
            id=""
            style="position: fixed; z-index: 1100; width: 300px; height: 100%; transform: translate3d(-100%, 0, 0); transition: all 0.5s; top: 0px; left: 0px; bottom: 0px;"
          >
            <div
              class="bm-menu"
              style="height: 100%; box-sizing: border-box; overflow: auto; background: rgb(39, 39, 39); padding: 1.5em 1em 0px; font-size: 1.15em;"
            >
              <nav
                class="bm-item-list"
                style="height: 100%; color: rgb(255, 255, 255); padding: 0.8em;"
              >
                <h3
                  class="sc-AykKF fCfuVZ bm-item"
                  style="display: block;"
                  tabindex="-1"
                >
                  Liste des chants
                </h3>
                <h3
                  class="sc-AykKF fCfuVZ bm-item"
                  style="display: block;"
                  tabindex="-1"
                >
                  Chant aléatoire
                </h3>
              </nav>
            </div>
            <div>
              <div
                class="bm-cross-button"
                style="position: absolute; width: 24px; height: 24px; right: 8px; top: 8px;"
              >
                <span
                  style="position: absolute; top: 6px; right: 14px;"
                >
                  <span
                    class="bm-cross"
                    style="position: absolute; width: 3px; height: 14px; transform: rotate(45deg); background: rgb(255, 255, 255);"
                  />
                  <span
                    class="bm-cross"
                    style="position: absolute; width: 3px; height: 14px; transform: rotate(-45deg); background: rgb(255, 255, 255);"
                  />
                </span>
                <button
                  style="position: absolute; left: 0px; top: 0px; width: 100%; height: 100%; margin: 0px; padding: 0px; font-size: 0; background: transparent; cursor: pointer;"
                  tabindex="-1"
                >
                  Close Menu
                </button>
              </div>
            </div>
          </div>
          <div>
            <div
              class="bm-burger-button"
              style="z-index: 1000; position: fixed; width: 36px; height: 30px; left: 16px; top: 16px;"
            >
              <span>
                <span
                  class="bm-burger-bars"
                  style="position: absolute; height: 20%; left: 0px; right: 0px; top: 0%; opacity: 1; background: rgb(255, 255, 255);"
                />
                <span
                  class="bm-burger-bars"
                  style="position: absolute; height: 20%; left: 0px; right: 0px; top: 40%; opacity: 1; background: rgb(255, 255, 255);"
                />
                <span
                  class="bm-burger-bars"
                  style="position: absolute; height: 20%; left: 0px; right: 0px; top: 80%; opacity: 1; background: rgb(255, 255, 255);"
                />
              </span>
              <button
                style="position: absolute; left: 0px; top: 0px; width: 100%; height: 100%; margin: 0px; padding: 0px; font-size: 0; background: transparent; cursor: pointer;"
              >
                Open Menu
              </button>
            </div>
          </div>
        </div>
        <h1
          class="sc-AykKH gJBOii"
        >
          Liste des chants
        </h1>
      </div>
      <div
        class="sc-AykKK dYxQsa"
      >
        <div
          class="sc-AykKE jOJAW css-2b097c-container"
        >
          <div
            class=" css-yk16xz-control"
          >
            <div
              class=" css-1hwfws3"
            >
              <div
                class=" css-1wa3eu0-placeholder"
              >
                Rechercher...
              </div>
              <div
                class="css-b8ldur-Input"
              >
                <div
                  class=""
                  style="display: inline-block;"
                >
                  <input
                    aria-autocomplete="list"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    id="react-select-2-input"
                    spellcheck="false"
                    style="box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                  <div
                    style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                  />
                </div>
              </div>
            </div>
            <div
              class=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class=" css-tlfecz-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-6q0nyr-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-fzXfLO iuSNjY"
        />
        <div
          class="sc-AykKC hvfnPU"
        >
          <p
            class="sc-AykKD cTiopU"
          >
            tata
          </p>
        </div>
        <div
          class="sc-AykKC hvfnPU"
        >
          <p
            class="sc-AykKD cTiopU"
          >
            toto
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AykKG eNPyVs"
    >
      <div>
        <div
          class="bm-overlay"
          style="position: fixed; z-index: 1000; width: 100%; height: 100%; background: rgba(0, 0, 0, 0.3); opacity: 0; transform: translate3d(100%, 0, 0); transition: opacity 0.3s, transform 0s 0.3s;"
        />
        <div
          class="bm-menu-wrap"
          id=""
          style="position: fixed; z-index: 1100; width: 300px; height: 100%; transform: translate3d(-100%, 0, 0); transition: all 0.5s; top: 0px; left: 0px; bottom: 0px;"
        >
          <div
            class="bm-menu"
            style="height: 100%; box-sizing: border-box; overflow: auto; background: rgb(39, 39, 39); padding: 1.5em 1em 0px; font-size: 1.15em;"
          >
            <nav
              class="bm-item-list"
              style="height: 100%; color: rgb(255, 255, 255); padding: 0.8em;"
            >
              <h3
                class="sc-AykKF fCfuVZ bm-item"
                style="display: block;"
                tabindex="-1"
              >
                Liste des chants
              </h3>
              <h3
                class="sc-AykKF fCfuVZ bm-item"
                style="display: block;"
                tabindex="-1"
              >
                Chant aléatoire
              </h3>
            </nav>
          </div>
          <div>
            <div
              class="bm-cross-button"
              style="position: absolute; width: 24px; height: 24px; right: 8px; top: 8px;"
            >
              <span
                style="position: absolute; top: 6px; right: 14px;"
              >
                <span
                  class="bm-cross"
                  style="position: absolute; width: 3px; height: 14px; transform: rotate(45deg); background: rgb(255, 255, 255);"
                />
                <span
                  class="bm-cross"
                  style="position: absolute; width: 3px; height: 14px; transform: rotate(-45deg); background: rgb(255, 255, 255);"
                />
              </span>
              <button
                style="position: absolute; left: 0px; top: 0px; width: 100%; height: 100%; margin: 0px; padding: 0px; font-size: 0; background: transparent; cursor: pointer;"
                tabindex="-1"
              >
                Close Menu
              </button>
            </div>
          </div>
        </div>
        <div>
          <div
            class="bm-burger-button"
            style="z-index: 1000; position: fixed; width: 36px; height: 30px; left: 16px; top: 16px;"
          >
            <span>
              <span
                class="bm-burger-bars"
                style="position: absolute; height: 20%; left: 0px; right: 0px; top: 0%; opacity: 1; background: rgb(255, 255, 255);"
              />
              <span
                class="bm-burger-bars"
                style="position: absolute; height: 20%; left: 0px; right: 0px; top: 40%; opacity: 1; background: rgb(255, 255, 255);"
              />
              <span
                class="bm-burger-bars"
                style="position: absolute; height: 20%; left: 0px; right: 0px; top: 80%; opacity: 1; background: rgb(255, 255, 255);"
              />
            </span>
            <button
              style="position: absolute; left: 0px; top: 0px; width: 100%; height: 100%; margin: 0px; padding: 0px; font-size: 0; background: transparent; cursor: pointer;"
            >
              Open Menu
            </button>
          </div>
        </div>
      </div>
      <h1
        class="sc-AykKH gJBOii"
      >
        Liste des chants
      </h1>
    </div>
    <div
      class="sc-AykKK dYxQsa"
    >
      <div
        class="sc-AykKE jOJAW css-2b097c-container"
      >
        <div
          class=" css-yk16xz-control"
        >
          <div
            class=" css-1hwfws3"
          >
            <div
              class=" css-1wa3eu0-placeholder"
            >
              Rechercher...
            </div>
            <div
              class="css-b8ldur-Input"
            >
              <div
                class=""
                style="display: inline-block;"
              >
                <input
                  aria-autocomplete="list"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  id="react-select-2-input"
                  spellcheck="false"
                  style="box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
                <div
                  style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                />
              </div>
            </div>
          </div>
          <div
            class=" css-1hb7zxy-IndicatorsContainer"
          >
            <span
              class=" css-1okebmr-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class=" css-tlfecz-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-6q0nyr-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-fzXfLO iuSNjY"
      />
      <div
        class="sc-AykKC hvfnPU"
      >
        <p
          class="sc-AykKD cTiopU"
        >
          tata
        </p>
      </div>
      <div
        class="sc-AykKC hvfnPU"
      >
        <p
          class="sc-AykKD cTiopU"
        >
          toto
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
